[core] Disable scissor test when using offscreen texture

brunoabinader opened the pull request:
When scissor test is enabled e.g. when embedding our engine on third-party platforms, the scissor test ends up polluting the rendering of offscreen textures e.g. used when rendering fill extrusion layers. This PR:
- Adds convenience methods for assuming/setting viewport values with non-zero `x` and `y` values
- Adds convenience methods for assuming/setting scissor test values
- Makes sure scissor test is disabled when using offscreen textures for fill extrusion layer

kkaefer commented on src/mbgl/renderer/painter.cpp:
```diff
@@ -327,6 +327,7 @@ void Painter::renderPass(PaintParameters& parameters,
 
             extrusionTexture->bind();
 
+            context.scissorTest = false;
```
Instead of setting this here, we should move that call into OffscreenTexture/OffscreenView's `bind` method, which is also where we call `context.viewport = { 0, 0, size };`

brunoabinader commented:
@kkaefer addressed review changes + updated `OffscreenTexture` tests to reflect these changes.

A custom layer that modifies any of the GL states not reset by `context.setDirtyState()` - like viewport and scissor - will cause unexpected behavior - because these changes won't be updated into our internal GL state manager.

In fact, we should either explicitly note that modifying such states inside a custom layer will cause unexpected behavior, or enforce that `Offscreen{View,Texture}::bind()` state resets are done by using `setCurrentValue` instead of `operator=` when custom layers are in use.
